### PR TITLE
gitlab: use urlparse instead of manually parsing the base_url

### LINKED ISSUE #2643: Unable to connect to repositories under GitLab subgroups
### Description of problem
When the depth of the repository has more than two levels, the kiwitcms is unable to communicate with GitLab

### Steps to Reproduce

1. Create a group on GitLab (G1)
2. Create a subgroup in it (G2)
3. Create a project in the subgroup (P1)
4.  Open a new issue on P1 
5. Setup the bug tracker as following:
* BASE_URL= https://gitlab.com/G1/G2/P1
* API_URL= https://gitlab.com
* username and password according to your settings
6. Health check:  https://gitlab.com/G1/G2/P1/-/issues/1

### Actual results
Two banners with failed communication (404 Can't find project) and the integration does not work 

### Expected results
It should work

### Additional info (Python traceback, logs, etc.)
I guess that the problem is in the GitlabThread class, in the repo_id method. 

`return "/".join(bug_system.base_url.strip().strip("/").split("/")[-2:])`

### PR DESCRIPTION
**Pull request by cmeissl**
`repo_id` currently always returns the last two segments of the `base_url`.
For projects under a sub group it removes parts of the group hierachy which results
in a `404` when trying to create a Issue from within `Kiwi`.

fixes #2643

### CODE REVIEW

File: tcms/issuetracker/gitlab_integration.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
     Helper which facilitates actual communications with GitLab.
 """
 from tcms.issuetracker.base import IntegrationThread
-
+from urllib.parse import urlparse
```
Comment (atodorov) [inline tcms/issuetracker/gitlab_integration.py:6]: see flake8, isort and black checks.

File: tcms/issuetracker/gitlab_integration.py [hunk 2]
```diff
@@ -14,7 +14,7 @@ class GitlabThread(IntegrationThread):
 
     @staticmethod
     def repo_id(bug_system):
-        return "/".join(bug_system.base_url.strip().strip("/").split("/")[-2:])
+        return urlparse(bug_system.base_url).path.lstrip('/')
```
[the remark this answers is not in the document]
Comment (atodorov) [reply tcms/issuetracker/gitlab_integration.py:17]: Looks like this is also causing the existing tests to fail.

**Review by atodorov — CHANGES_REQUESTED** [review]
Needs some fixes.

Also needs an additional test for the case where project name has 3 components separated by a slash, see `tests/gitlab_ce/initialize-data` and `tcms/issuetracker/tests/test_gitlab_ee.py`.

File: tcms/issuetracker/gitlab_integration.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (cmeissl) [reply tcms/issuetracker/gitlab_integration.py]: right, thanks, should be fixed now

**Comment by cmeissl** [comment]
> Needs some fixes.
> 
> Also needs an additional test for the case where project name has 3 components separated by a slash, see `tests/gitlab_ee/initialize-data` and `tcms/issuetracker/tests/test_gitlab_ee.py`.

Thanks for pointing me into the right direction. I have extended `initialize-data` to create the hierachy for testing projects in groups `group->sub_group->kiwitcms_in_group`. 

The gitlab tests have been extended with:
- `test_bug_id_from_url_in_group`
- `test_details_for_public_url_in_group`

The tests in the CI are now green again for gitlab, but some other unrelated tests seem to fail, at least I fail to see how
this PR could have broken the other tests.

**Pull request merged**